h/services/user.py

```diff
@@ -147,7 +147,7 @@ def fetch_for_login(self, username_or_email):
         :rtype: h.models.User or NoneType
         :raises UserNotActivated: When the user is not activated.
         """
-        filters = [(User.authority == self.default_authority)]
+        filters = [(User.authority == self.default_authority), User.deleted.is_(False)]
```

Comment:
Prevent `UserService.fetch_for_login()` from returning users that've been deleted. This results in a "User does not exist" error if you try to log in as this user.

Other `UserService` methods still continue to return deleted users, so other functionality should be unaffected.